### ISSUE TITLE
Rule Change[no-missing-label-refs]: support GFM alerts

### ISSUE THREAD

**Issue by DMartens** [eslint/markdown#310]
### What rule do you want to change?

no-missing-label-refs

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```markdown
> [!NOTE]
> Description
```

### What does the rule currently do for this code?

The rule uses a regular expression to find missing labels but does not ignore [GFM-style alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).

### What will the rule do after it's changed?

Ignore the alerts (e.g. `[!NOTE]`) as they are not a link reference.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

It seems that rules have no access to which language is used and as such the change would also affect the `commonmark` mode.
